### PR TITLE
[DPE-2467] Socket connection

### PR DESCRIPTION
**Pull request by dragomirp**
## Issue
Default configuration prevents socket connections for the operator user from within the charm machine

## Solution
* Allow socket connections
* Restrict the access to Patroni configuration

### CODE REVIEW

File: tests/unit/test_cluster.py [hunk 1]
```diff
@@ -273,7 +273,7 @@ def test_render_patroni_yml_file(self, _, _render_file, _get_postgresql_version)
         _render_file.assert_called_once_with(
             "/var/snap/charmed-postgresql/current/etc/patroni/patroni.yaml",
             expected_content,
-            0o644,
+            0o600,
```
Comment (dragomirp) [inline tests/unit/test_cluster.py:276]: Should we keep the group permissions here?

File: templates/patroni.yml.j2 [hunk 2]
```diff
@@ -99,6 +99,7 @@ postgresql:
   pgpass: /tmp/pgpass
   pg_hba:
     - local all backup peer map=operator
+    - local all operator password
```
Comment (dragomirp) [inline templates/patroni.yml.j2:102]: This is plaintext password auth. It should be fine for socket connections and in line with how monitoring connects. Alternatively we can demand md5 or scram.

File: tests/unit/test_cluster.py (hunk 1 above)
Comment (marceloneppel) [reply tests/unit/test_cluster.py:276]: From what I remember, we don't need that.

**Review by marceloneppel — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]
Tnx! BTW, are you going to allow socket connection for K8s as well?

**Pull request merged**